Sources/GRPCHTTP2Core/Server/HTTP2ServerTransport.swift

```diff
@@ -188,4 +188,142 @@ extension HTTP2ServerTransport.Config {
       Self(maxRequestPayloadSize: 4 * 1024 * 1024)
     }
   }
+
+  public struct TransportSecurity: Sendable {
+    private enum Wrapped {
+      case plaintext
+      case tls(TLS)
+    }
+
+    private let wrapped: Wrapped
+
+    /// This connection is plaintext: no encryption will take place.
+    public static let plaintext = Self(wrapped: .plaintext)
+
+    /// This connection will use TLS.
+    public static func tls(_ tls: TLS) -> Self {
+      Self(wrapped: .tls(tls))
+    }
+
+    /// Returns the TLS configuration, if the security has been set to ``tls(_:)``.
+    package var tlsConfig: TLS? {
+      switch wrapped {
+      case .plaintext:
+        return nil
+      case .tls(let config):
+        return config
+      }
+    }
+  }
+
+  public struct TLS: Sendable {
+    /// The serialization format of the provided certificates and private keys.
+    public struct SerializationFormat: Sendable, Equatable {
+      package enum Wrapped {
+        case pem
+        case der
+      }
+
+      package let wrapped: Wrapped
+
+      public static let pem = Self(wrapped: .pem)
+      public static let der = Self(wrapped: .der)
+    }
+
+    public struct CertificateSource: Sendable {
+      package enum Wrapped {
+        case file(path: String, serializationFormat: SerializationFormat)
+        case bytes(bytes: [UInt8], serializationFormat: SerializationFormat)
+      }
+
+      package let wrapped: Wrapped
+
+      /// The certificate will be provided via a file.
+      public static func file(path: String, serializationFormat: SerializationFormat) -> Self {
+        Self(wrapped: .file(path: path, serializationFormat: serializationFormat))
+      }
+
+      /// The certificate will be provided as an array of bytes.
+      public static func bytes(
+        _ bytes: [UInt8],
+        serializationFormat: SerializationFormat
```

Comment:
I think we can drop 'serialization' here and in other places (it's less verbose and doesn't add any ambiguity)
```suggestion
        format: SerializationFormat
```